File: flint/src/jp/oist/flint/filesystem/Filename.java
/* -*- Mode: Java; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:set ts=4 sw=4 sts=4 et: */
package jp.oist.flint.filesystem;

import java.io.File;

public class Filename {

    public static String getNameWithoutExtension(File file) {
        String fileName = file.getName();
        int postion = fileName.lastIndexOf('.');
        if (postion == -1) return fileName;
        return fileName.substring(0, postion);
    }

    public static File getVariant(File sourceFile, File targetDir, int index, int numDigits, String extension) {
        String name = getNameWithoutExtension(sourceFile);
        String format = "%s_%0" + numDigits + "d.%s";
        String basename = String.format(format, name, index, extension);
        return new File(targetDir, basename);
    }
}

File: source/src/gui/filename.cc
/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: t; c-basic-offset: 4 -*- vim:set ts=4 sw=4 sts=4 noet: */
#ifdef HAVE_CONFIG_H
#include "config.h"
#endif

#include "gui/filename.h"

#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wcast-qual"
#include <wx/wx.h>
#pragma GCC diagnostic pop

#include <cassert>

namespace flint {
namespace gui {

void AppendCurrentTimestampDir(wxFileName &fileName)
{
	auto now = wxDateTime::Now();
	bool b = fileName.AppendDir(now.Format("%Y%m%d-%H%M%S"));
	assert(b);
}

std::string GetFnStrFromWxFileName(const wxFileName &fileName)
{
	wxString fullPath = fileName.GetFullPath();
	return fullPath.ToStdString(); // TODO: check locale-dependency
}

boost::filesystem::path GetPathFromWxFileName(const wxFileName &fileName)
{
	wxString fullPath = fileName.GetFullPath();
	return GetPathFromWxString(fullPath);
}

boost::filesystem::path GetPathFromWxString(const wxString &str)
{
	boost::filesystem::path path(str.fn_str()); // TODO: check locale-dependency
	return path;
}

}
}
